systest/backup/nfs-backup/backup_test.go

```diff
@@ -0,0 +1,326 @@
+	"github.com/dgraph-io/dgraph/x"
+
+	"github.com/dgraph-io/dgo/v210"
+	"github.com/dgraph-io/dgo/v210/protos/api"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+)
+
+var (
+	copyBackupDir  = "./data/backups_copy"
+	backupDir      = "./data/backup"
+	restoreDir     = "./data/restore"
+	testDirs       = []string{backupDir, restoreDir}
+	backupDstHA    = "/ha_backup"
+	backupDstNonHA = "/non_ha_backup"
+)
+
+func TestBackupHAClust(t *testing.T) {
+
+	backupRestoreTest(t, testutil.SockAddr, testutil.SockAddrAlpha4Http, testutil.SockAddrZeroHttp, backupDstHA, testutil.SockAddrHttp)
+}
+
+func TestBackupNonHAClust(t *testing.T) {
+
+	backupRestoreTest(t, testutil.SockAddrAlpha7, testutil.SockAddrAlpha8Http, testutil.SockAddrZero7Http, backupDstNonHA, testutil.SockAddrAlpha7Http)
+}
+
+func backupRestoreTest(t *testing.T, backupAlphaSocketAddr string, restoreAlphaAddr string, backupZeroAddr string, backupDst string, backupAlphaSocketAddrHttp string) {
+	conn, err := grpc.Dial(backupAlphaSocketAddr, grpc.WithInsecure())
+	require.NoError(t, err)
+	dg := dgo.NewDgraphClient(api.NewDgraphClient(conn))
+	ctx := context.Background()
+	require.NoError(t, dg.Alter(ctx, &api.Operation{DropAll: true}))
+	// Add schema and types.
+	require.NoError(t, dg.Alter(ctx, &api.Operation{Schema: `movie: string .
+		 type Node {
+			 movie
+		 }`}))
+
+	// Add initial data.
+	original, err := dg.NewTxn().Mutate(ctx, &api.Mutation{
+		CommitNow: true,
+		SetNquads: []byte(`
+			 <_:x1> <movie> "BIRDS MAN OR (THE UNEXPECTED VIRTUE OF IGNORANCE)" .
+			 <_:x2> <movie> "Spotlight" .
+			 <_:x3> <movie> "Moonlight" .
+			 <_:x4> <movie> "THE SHAPE OF WATERLOO" .
+			 <_:x5> <movie> "BLACK PUNTER" .
+		 `),
+	})
+	require.NoError(t, err)
+	t.Logf("--- Original uid mapping: %+v\n", original.Uids)
+	// Move tablet to group 1 to avoid messes later.
+	// client := *http.DefaultClient
+	_, err = http.Get("http://" + backupZeroAddr + "/moveTablet?tablet=movie&group=1")
+	require.NoError(t, err)
+	// After the move, we need to pause a bit to give zero a chance to quorum.
+	t.Log("Pausing to let zero move tablet...")
+	moveOk := false
+	for retry := 5; retry > 0; retry-- {
+		state, err := testutil.GetState()
+		require.NoError(t, err)
+		if _, ok := state.Groups["1"].Tablets[x.NamespaceAttr(x.GalaxyNamespace, "movie")]; ok {
+			moveOk = true
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+	require.True(t, moveOk)
+	// Setup test directories.
+	dirSetup(t)
+	// Send backup request.
+	//       mostly because of a race condition
+	//       adding sleep
+	time.Sleep(time.Second * 10)
+	_ = runBackup(t, 1, 1, backupAlphaSocketAddrHttp, backupDst)
+	restoreStatusCode := runRestore(t, "", restoreAlphaAddr, backupDst, false, 1)
+	require.Equal(t, "Success", restoreStatusCode)
+	testutil.WaitForRestore(t, dg, restoreAlphaAddr)
+	// We check expected Objects vs Received Objects from restoreed db
+	checkObjectCount(t, 5, 5, restoreAlphaAddr)
```

Comment:
Rather than hard coding numbers like `5`, can we give them appropriate variable name and use that here instead?

Reply:
done @anurags92